Automattic/wp-calypso
# State: Implement page templates state

Related: #5785 (cc @jeremeylduvall)

This pull request seeks to implement the `pageTemplates` Redux state, currently including actions and reducers for requesting and tracking page templates. This does not yet include any revisions to the application to make use of this state, nor does it include a query component for triggering the request.

This state will eventually be used as part of the Reduxification effort in the post editor.

It does, however, include a small revision to the `createReducer` utility to throw an error in non-production environments when the reducer is called with a `type` value of `undefined`. While implementing this subtree, I had initially forgotten to define the action types, and the errors I encountered in my tests were very non-obvious. This should help a bit for developing new state subtrees. (cc @gziolo)

__Testing instructions:__

There should be no changes to the application. Verify that Calypso loads as expected, and that Mocha tests pass:

```
npm run test-client client/state
```

Test live: https://calypso.live/?branch=add/state-page-templates
